Fix updated Euler angle computation for crystal plasticity models (#19494)
idaholab/moose: modules/tensor_mechanics/doc/content/source/materials/crystal_plasticity/ComputeUpdatedEulerAngle.md

Below are the inverse pole figures for the three cases. It can be seen that the $[111]$ oriented crystal pole does not move in the inverse pole figure during deformation.  A perfectly oriented $[112]$ is initially situated on the symmetry line in the inverse pole figure. When this crystal is pulled, the pole moves along the symmetry line towards $[111]$. For the $[123]$ crystal, the crystal tensile axis is initially inside the inverse pole figure and moves towards $[111]$ when deformed in tension.

!media tensor_mechanics/crystal_plasticity/Inverse_Pole_Fig.png
    id=IPF_Orients
    caption=Inverse pole figure of a singe crystal undergone uniaxial tension. Three cases with different initial orientations are demonstrated. The initial orientations are $[111]$, $[112]$, and $[123]$ from the left to the right.
    style=display:block;margin-left:auto;margin-right:auto;width:90%

> The inverse pole figures are plotted using [MTEX](https://github.com/mtex-toolbox/mtex) under matlab. One can reproduce the data that correspond to the above results by running [this test](modules/tensor_mechanics/test/tests/crystal_plasticity/stress_update_material_based/update_euler_angle.i) with a larger end_time (50 seconds).

Reviewer: ```suggestion
The inverse pole figures are plotted using MATLAB tool box [MTEX](https://github.com/mtex-toolbox/mtex). The above results can be reproduced  by running [this test](modules/tensor_mechanics/test/tests/crystal_plasticity/stress_update_material_based/update_euler_angle.i) with  end_time = 50 (seconds).
```

> The MOOSE developer would like to acknowledge Dr. Alankar Alankar and Dr. Ritam Chatterjee's contributions in this verification case.

Reviewer: ```suggestion
The MOOSE developer would like to acknowledge Dr. Alankar Alankar and Ritam Chatterjee's contributions to this verification case.
```
